ci_framework/roles/edpm_prepare/tasks/main.yml

```diff
@@ -0,0 +1,98 @@
+      OUT: {{ cifmw_edpm_prepare_intall_yamls_output_dir }}
+    cifmw_edpm_prepare_make_openstack_env: |
+      {% if cifmw_operator_build_meta_name in cifmw_operator_build_output.operators %}
+      OPENSTACK_IMG: {{ cifmw_operator_build_output.operators[cifmw_operator_build_meta_name].image }}
+      {% endif %}
+    cifmw_edpm_prepare_make_openstack_deploy_env: |
+      {% if cifmw_operator_build_meta_name in cifmw_operator_build_output.operators %}
+      OPENSTACK_BRANCH: ""
+      GIT_CLONE_OPTS: "-l"
+      OPENSTACK_REPO: cifmw_operator_build_output.operators[cifmw_operator_build_meta_name].git_src_dir
+      {% endif %}
+
+- name: Prepare inputs
+  vars:
+    make_input_env: "{{ cifmw_edpm_prepare_common_env | from_yaml }}"
+  ansible.builtin.include_role:
+    name: 'install_yamls_makes'
+    tasks_from: 'make_input'
+
+# TODO: Prepare a proper operator list that dictates what operators to deploy instead of relaying on a simple flag
+- name: OpenStack meta-operator installation
+  when: not cifmw_edpm_skip_openstack_operator
+  block:
+  - name: Install OpenStack operator
+    vars:
+      make_openstack_env: "{{ cifmw_edpm_prepare_common_env | from_yaml |
+        combine(cifmw_edpm_prepare_make_openstack_env | from_yaml)}}"
+    ansible.builtin.include_role:
+      name: 'install_yamls_makes'
+      tasks_from: 'make_openstack'
+
+  - name: Wait for OpenStack subscription creation
+    ansible.builtin.shell: |
+      oc get sub openstack-operator --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }} > /dev/null 2>&1 &&
+        oc get sub openstack-operator --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }} -o=jsonpath='{.status.installplan.name}'
+    until: cifmw_edpm_prepare_wait_installplan_out.rc == 0 and cifmw_edpm_prepare_wait_installplan_out.stdout != ""
+    register: cifmw_edpm_prepare_wait_installplan_out
+    retries: 30
+    delay: 5
+
+  - name: Wait for OpenStack operator to get installed
+    ansible.builtin.command:
+      cmd: >
+        oc wait InstallPlan {{ cifmw_edpm_prepare_wait_installplan_out.stdout }}
+        --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }} --for=jsonpath='{.status.phase}'=Complete --timeout=20m
+
+- name: Install OpenStack service
+  vars:
+    make_openstack_deploy_env: "{{ cifmw_edpm_prepare_common_env | from_yaml |
+      combine(cifmw_edpm_prepare_make_openstack_deploy_env | from_yaml)}}"
+  ansible.builtin.include_role:
+    name: 'install_yamls_makes'
+    tasks_from: 'make_openstack_deploy'
+
+- name: Find the OpenStack CR manifest
+  ansible.builtin.find:
+    paths: "{{ cifmw_edpm_prepare_intall_yamls_output_dir }}/{{ cifmw_install_yamls_defaults['NAMESPACE'] }}/openstack/cr"
+    contains: "kind: OpenStackControlPlane"
+    patterns: "*.yaml"
+  register: cifmw_edpm_prepare_openstack_cr_manifest_paths
+
+- name: Ensure manifest exists
+  ansible.builtin.assert:
+    that: cifmw_edpm_prepare_openstack_cr_manifest_paths.matched == 1
```

Comment:
I think we may have to check "> 0" instead here - we may end with custom files in that location, and they may match the same kind. In case we're updating it later... ?

Reply:
Honestly, I don't like the whole solution, is something that works for a first iteration, but I'll probably create a custom Python module that given a path is able to apply the kustomization and ouput the resulting manifests, with maybe the ability to retrieve only a field given a yaml path. I put 1 as value for the assert cause I expect to have a single control plane CR applied, if there a more than one that's something that I'm not expecting, and in fact, I'll wait only for one control plane.